### DISCUSSION TITLE
解决Google账户登录点下一点按钮无反应的问题

### DISCUSSION THREAD

**Discussion by elfzweik** [XTLS/Xray-core#1618] [Show and tell]
最近国内朋友反映用我的xray服务器fq时，无法添加Google账户，具体表现为在输入邮箱地址后，点下一步无反应。
用edge或chrome翻墙都要用到SwitchyOmega，在点了下一步按钮时，发现SwitchyOmega的蓝圈圈上出现了1个小字4，说明有4个网站的信息没有获得，点进去看一下，是*.gstatic.com，因此怀疑是某一个gstatic.com作的怪。gstatic.com是谷歌广告的网站记数服务器，在国内有服务器，因此按照一般的xray分流策略，gstatic.com不会走proxy出口，而是走直连。这是没有问题的，问题是SwitchyOmega可能把gstatic.com解析到国外去了，所以造成了gstatic.com的资源无法获取。
进一步查资料和尝试，发现点击谷歌账号登录下一步时，需要访问ssl.gstatic.com。所以只要把ssl.gstatic.com放入SwitchyOmega的bypass list里就解决了。

**Comment by Extreme-Icer** [comment]
gstatic原则上不是不走境内才能被链接么

**Reply by SakuraSakuraSakuraChan** [reply]
> gstatic原则上不是不走境内才能被链接么

不是，比如fonts和ssl在国内都有服务器，但是有少数地方的DNS会解析到国外(被墙的谷歌IP)去

**Reply by luis1111111111** [reply]
咕咕等这些被重点关注、服务功能受地理位置限制的公司的各种域名，直连、使用其国内服务器，这不好吧？
一方面连接稳定性不会好吧？
另外地理位置判断上有隐患。解决地理位置判断问题，也是这些软件、服务的存在意义之一，不是吗？

再有，不是有专门的规则集只代理被阻断干扰的地址（无论国内外默认全部直连，然后只有被阻断的地址代理），有需要的去用那个规则集不就好了吗，为什么要在白名单里把gstatic这类国外地址加到cn里？

**Comment by ACodeHX** [comment]
可以试试将代理改成全局的

**Reply by AbnerAI** [reply]
No

**Reply by bryanwcao-create** [reply]
太优秀

**Comment by Benjamin1919** [comment]
直接把geoip:google和geosite:google全都代理算了，注意是两条规则，别把这俩合成一条规则。
国内网络污染严重，只能这样了

**Reply by luis1111111111** [reply]
不光是咕咕的域名，还有好多其他国外域名也在cn里

**Comment by PingZi-Wing** [comment]
感谢大佬的解决方案。我是在第三方使用谷歌账号登录时选择账号没反应，ssl.gstatic.com走了代理不行，让它走直连才登录成功，太迷了。

**Reply by luis1111111111** [reply]
你这个是服务器设置的问题吧，最近用白名单（默认全代理，国内地址直连）发现gstatic是直连状态连不上，用全局就没问题。

**Reply by Ai-Violet** [reply]
确实全局代理就可以

**Comment by itheabyss** [comment]
谢谢，今天我也出现这个情况，弄了一个小时，莫名其妙，最终搜到这个，一下就解决了。感谢！！！

**Comment by AmirS666** [comment]
Well, I had the same issue even when accessing different services like iMessage and... . I use Warp+, and I thought that this was the issue, so I tried to make a list for routing directly above other rules. I use the below domains, and the problem has been solved. 

`geosite:google,geosite:google-ads,geosite:google-trust-services,geosite:appleinsider,geosite:apple,geosite:appledaily,geosite:apple-update,geosite:apple-pki,geosite:apple-ads,geosite:apple-dev,accounts.google.com,mail.google.com,keyword:google,drive.google.com,keyword:apple,keyword:appleid,keyword:appe-store,keyword:appstore,keyword:appleappstore,keyword:gstatic,com.apple.MobileSMS,keyword:MobileSMS,keyword:apple.metal,geosite:google,geosite:apple,geosite:appledaily,geosite:apple-update,geosite:apple-pki,geosite:apple-ads,geosite:apple-dev,accounts.google.com,mail.google.com,keyword:google,drive.google.com,keyword:apple,keyword:appleid,keyword:appe-store,keyword:appstore,keyword:appleappstore,keyword:gstatic,init-p01md.apple.com,prod.apple.com,captive.apple.com,gsa.apple.com,pages.store.office.com,init.itunes.apple.com,fpinit.itunes.apple.com,setup.icloud.com,g.live.com,bag.itunes.apple.com,www.kte.mifi.com`

**Comment by xlsqac** [comment]
把 gstaic 设置直连就能解决

**Comment by CarpCap** [comment]
谢谢，确实把ssl.gstatic.com改为直链 就可以解决了

**Comment by jy0001087** [comment]
还是手机浏览器装了SwitchyOmega ，选择全局代理，解决问题

**Comment by Siriuya** [comment]
用另一种方式解决了。修改了一下hosts文件。
![photo_2024-10-23_10-23-06](https://github.com/user-attachments/assets/e19dc279-356e-4460-b496-a9fe2bd5a2c2)

**Comment by qhghong** [comment]
解决了，感谢

**Comment by kotarou5106** [comment]
开全局弄好了

**Comment by snhun** [comment]
正解，烦恼了许久！

**Comment by lory-lzj** [comment]
请问怎么把把ssl.gstatic.com放入SwitchyOmega的bypass list里呀？有操作步骤嘛？（电脑小白）

**Reply by ykfq** [reply]
<img width="777" height="247" alt="image" src="https://github.com/user-attachments/assets/aa9e240a-79d8-42cf-a067-f9acd1bb5cae" />
在 **情景模式：auto switch** 中将这几个域名添加进去就行了

**Comment by xhzcool** [comment]
Sign in一直加载怎么办？Google OAuth 登录的二次验证页面，显示的是：

“Make sure that you downloaded this app from Google”

以及下面的 Sign in / Cancel 按钮。

**Reply by ZhWang1104** [reply]
你找到什么解决办法了吗我也遇到这种问题了

**Comment by edisoncomeon** [comment]
用的时候发现了个问题，前面用的好好的，突然就节点全部超时，用不了了。如果发生这种情况，切换到【配置】界面，然后把直连DNS改成119.29.29.29（默认是1.1.1.1）。手机版也是一样的操作。

**Reply by 8181** [reply]
原来用的8.8.8.8，应该是被污染了，换成119.29.29.29这个就可以了

**Comment by panyuhang93** [comment]
试了一下,只要让ssl.gstatic.com直连就行了,其他方法果然没什么效果,太感谢了

**Comment by satangele** [comment]
主要原因应该还是服务端禁回国流量了，在配置文件路由最前面把geosite:google和geosite:google-cn放行应该就行

**Comment by benjinXie** [comment]
牛逼！我重装系统后发现死活都无法登陆，按照你的方法把这个网站标为直连后就成功了

**Comment by drawfromwreck** [comment]
你真是天才！我把ssl那个走直接连接就行了！ 我就说为什么这阵子登录google都怪怪的

**Comment by bytecategory** [comment]
nameserver 1.1.1.1

**Comment by orzuzro** [comment]
同样的问题，已经折腾了几周了。配置：macOS v2rayN vless reality vision 代理网络正常，登录gmail页面时点击按钮没有反应，浏览器F12后台日志 GET gstatic.com ERR_CONNECTION_CLOSE 改了楼主的规则 还是不行，只能来问问大家 还有其它的方法吗？

**Reply by orzuzro** [reply]
更新：重新整理了下v2rayN 的routing setting 把楼主的直连域名加进去，网络瞬间通畅了。我还有一个困扰很久的问题，打开 apple store 页面没有任何东西，一片灰，n2rayN 日志显示  *.apple.com socks-proxy 代理是通的，这个问题是哪个环节出错了？楼主能否提点下  @elfzweik   楼主配享太庙~~
